src/pages/images/actions/UploadInstanceBtn.tsx

```diff
@@ -0,0 +1,252 @@
+import { useProject } from "context/project";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { uploadInstance } from "api/instances";
+import { UploadState } from "types/storage";
+import { useEventQueue } from "context/eventQueue";
+import { useToastNotification } from "context/toastNotificationProvider";
+import ProgressBar from "components/ProgressBar";
+import { humanFileSize } from "util/helpers";
+import usePortal from "react-useportal";
+import StoragePoolSelector from "pages/storage/StoragePoolSelector";
+import { AxiosError } from "axios";
+import { LxdSyncResponse } from "types/apiResponse";
+
+export interface UploadInstanceFormValues {
+  instanceFile: File | null;
+  name: string;
+  pool: string;
+}
+
+const UploadInstanceBtn: FC = () => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const { project, isLoading: isProjectLoading } = useProject();
+  const [isUploading, setIsUploading] = useState(false);
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const [uploadAbort, setUploadAbort] = useState<AbortController | null>(null);
+  const instanceNameAbort = useState<AbortController | null>(null);
+
+  const changeFile = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("instanceFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const suffix = "-imported";
+      const instanceName = truncateInstanceName(
+        // remove file extension
+        file.name.split(".tar")[0],
+        suffix,
+      );
+      await formik.setFieldValue("instanceFile", file);
+      await formik.setFieldValue("name", instanceName);
+
+      // validate instance name
+      await formik.validateField("name");
+      void formik.setFieldTouched("name", true, true);
+      if (!formik.errors.name) {
+        formik.setFieldError("name", undefined);
+      }
+    }
+  };
+
+  const handleCloseModal = useCallback(() => {
+    uploadAbort?.abort();
+    setIsUploading(false);
+    setUploadState(null);
+    setUploadAbort(null);
+    formik.resetForm();
+    closePortal();
+  }, [uploadAbort]);
+
+  const onSuccess = (instanceName: string) => {
+    const message = (
+      <>
+        Created Instance <strong>{instanceName}</strong>.
+      </>
+    );
+
+    const actions = [
+      {
+        label: "Configure",
+        onClick: () =>
+          navigate(
+            `/ui/project/${project?.name}/instance/${instanceName}/configuration`,
+          ),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const onFailure = (msg: string) => {
+    toastNotify.failure("Instance creation failed.", new Error(msg));
+  };
+
+  const onFinish = () => {
```

Comment:
```suggestion
  const handleFinish = () => {
```
I think local functions that are called should be named handleFoo, only the received callbacks, that are called on specific events are named onFoo. Same for the onFailure and onSuccess above.